src/Services/ResultService.php

```diff
@@ -122,4 +104,29 @@ private function getAnswerForQuestionId(string $getId): ?TextInstance
             return null;
         }
     }
-}
\ No newline at end of file
+
+    private function getQuizInstanceDTOsWithOffset(string $offset): array
+    {
+        $resultRepository = $this->repositoryManager->getRepository(QuizInstance::class);
+        $quizInstances = $resultRepository->findBy([], [], 10, $offset);
+
+        $quizInstanceDTOs = [];
+        $userRepository = $this->repositoryManager->getRepository(User::class);
+        foreach ($quizInstances as $quizInstance) {
+            $userEntity = $userRepository->find($quizInstance->getUserId());
```

Comment:
you can think about a small "caching" mechanism here if you want. you might perform this query too many times with the same id